derotation/analysis/full_rotation_pipeline.py

```diff
@@ -588,29 +561,28 @@ def remove_artifacts_from_interpolated_angles(self):
         """
         logging.info("Cleaning interpolated angles...")
 
-        # find very short rotation periods in self.interpolated_angles
-
         self.config["analog_signals_processing"][
             "angle_interpolation_artifact_threshold"
         ]
         thresholded = np.zeros_like(self.interpolated_angles)
         thresholded[np.abs(self.interpolated_angles) > 0.15] = 1
-        rotation_start = np.nonzero(np.diff(thresholded) > 0)[0]
-        rotation_end = np.nonzero(np.diff(thresholded) < 0)[0]
+        rotation_start = np.where(np.diff(thresholded) > 0)[0]
+        rotation_end = np.where(np.diff(thresholded) < 0)[0]
 
-        self.check_rotation_number(start=rotation_start, end=rotation_end)
+        self.check_rotation_number_after_interpolation(
+            rotation_start, rotation_end
+        )
 
-        for i, (start, end) in enumerate(
-            zip(rotation_start[1:], rotation_end[:-1])
-        ):
+        for start, end in zip(rotation_start[1:], rotation_end[:-1]):
             self.interpolated_angles[end:start] = 0
```

Comment:
Yes that's what it is 😅 do you find it confusing?

Reply:
I don't think its so confusing and makes sense in the context, but for a reader not particularly familiar with the procedure like myself it might be a little unexpected (i.e. just indexing an array `end:start`). Maybe just a brief clarifying comment above the line? or maybe the variables could be `last_rotation_end`, `next_rotation_start` if not too verbose.